### DISCUSSION TITLE
Please remove 'For you'

### DISCUSSION THREAD

**Discussion by DarioKolic** [community/community#13395] [Other Feature Feedback, Questions, & Ideas]
This is not a social media platform and please do not try to make it one.

**Comment by svalordev** [comment]
I honestly think GitHub should drop the whole feed thing. Maybe we could have a dashboard at its place where it shows all our repositories stats or something like that.

**Comment by KSH-code** [comment]
Even the information is not helpful.

**Reply by baggiponte** [reply]
yeah i scrolled this for you feed and found it just confusing and chaotic. my following page, combined with trends, is already great and I could find many interesting new libraries, updates to existing projects, and new people to follow.

**Comment by Decode84** [comment]
I admire that Microsoft is trying to connect githubs users with each other, so you can discover repositories easier.

**Comment by tlebeitsuk** [comment]
I like it!

**Reply by Pingger** [reply]
I don't. It keeps me an additional click away from the repos I'm following, while not showing anything that interests me. There is not a single thing from the repos I'm following. and believe you me, there is not a lack of things that could be displayed there. And in addition: Several repos I'm watching now no longer show up in the following section!

**Comment by ghost** [comment]
I think the old design was simpler and more beautiful than this, with many additions, showing some of the bugs

**Comment by jfullstackdev** [comment]
hello guys, there is a new update on the For You feed, you can now filter it, and for those who don't want it can simply disable it by unchecking all filters

![image](https://user-images.githubusercontent.com/47092464/182012117-0b19be9d-e849-462e-909e-d20d28915c1d.png)

**Comment by NitinIsFit** [comment]
If you want them to remove for you page, then simply just don't open the page and ignore it? ¯\_(ツ)_/¯

**Reply by JoshuaAmaju** [reply]
I don't understand the outrage, like if you don't like it don't click it. But I guess people don't like change and like to impose their preferences.

**Reply by Pingger** [reply]
The issue is, that they broke the old homepage. It no longer shows the information it did previously! In addition, everytime I go to 'github.com' I end up on that new 'For You'-Page and thus am a click further away from the old homepage.

The 'Following'-Tab (old Homepage) is described as follows:

> The feed shows you events from people you follow and repositories you watch or star.

Since the new 'For You'-Page exists, it now only shows change to a single repository by a single user. changes by other users or in other repositories are not displayed at all.


**And this is why people are so outraged!** The issue isn't the new feature. The issue is, that they broke the old one.

**Reply by NitinIsFit** [reply]
> I don't understand the outrage, like if you don't like it don't click it. But I guess people don't like change and like to impose their preferences.

Everything you will see on the screen is the "for you" button, you just have to ignore it and not to click on it, it's negligible.

**Reply by NitinIsFit** [reply]
> The issue is, that they broke the old homepage. It no longer shows the information it did previously! In addition, everytime I go to 'github.com' I end up on that new 'For You'-Page and thus am a click further away from the old homepage.
> 
> The 'Following'-Tab (old Homepage) is described as follows:
> 
> > The feed shows you events from people you follow and repositories you watch or star.
> 
> Since the new 'For You'-Page exists, it now only shows change to a single repository by a single user. changes by other users or in other repositories are not displayed at all.
> 
> 
> **And this is why people are so outraged!** The issue isn't the new feature. The issue is, that they broke the old one.

Sounds like problem, oof
But can you tell what changes did you notice before after the creation of for you page, like

> The issue is, that they broke the old homepage. It no longer shows the information it did previously!

What information?

**Reply by NitinIsFit** [reply]
The "following" page is still here

**Reply by Pingger** [reply]
> What information?

Previously these were displayed (at least for me):
- New forks and projects by Users I'm following.
- New commits, releases, branches, issues, pull requests, and more for repositories I'm watching or have stared
- New Releases for repos in which I only have selected 'Releases' in the watch options
- New comments to issues/pull requests in which I participate.
- Any comments where I'm mentioned

All Completely gone.

I now only have entries for commits by a singular user, that I'm not following, for a singular repo, that I'm following.
That users also commits ot other repos that I'm following, but those are not displayed. Also other users commit to that first repo, which also no longer show up.

Comments, Issues, branches, releases and Pull Requests in any form are entirely gone from the old 'Following'-Tab.

PS: **Even activity in my own repositories is entirely gone**

**Reply by NitinIsFit** [reply]
> > What information?
> 
> Previously these were displayed (at least for me):
> - New forks and projects by Users I'm following.
> - New commits, releases, branches, issues, pull requests, and more for repositories I'm watching or have stared
> - New Releases for repos in which I only have selected 'Releases' in the watch options
> - New comments to issues/pull requests in which I participate.
> - Any comments where I'm mentioned
> 
> All Completely gone.
> 
> I now only have entries for commits by a singular user, that I'm not following, for a singular repo, that I'm following.
> That users also commits ot other repos that I'm following, but those are not displayed. Also other users commit to that first repo, which also no longer show up.
> 
> Comments, Issues, branches, releases and Pull Requests in any form are entirely gone from the old 'Following'-Tab.
> 
> PS: **Even activity in my own repositories is entirely gone**
> 

@Pingger Thanks for elaborating, i think we gotta make a separate post for this issue.

**Reply by NitinIsFit** [reply]
@Pingger https://github.com/orgs/community/discussions/27818

**Reply by NitinIsFit** [reply]
> > What information?
> 
> Previously these were displayed (at least for me):
> - New forks and projects by Users I'm following.
> - New commits, releases, branches, issues, pull requests, and more for repositories I'm watching or have stared
> - New Releases for repos in which I only have selected 'Releases' in the watch options
> - New comments to issues/pull requests in which I participate.
> - Any comments where I'm mentioned
> 
> All Completely gone.
> 
> I now only have entries for commits by a singular user, that I'm not following, for a singular repo, that I'm following.
> That users also commits ot other repos that I'm following, but those are not displayed. Also other users commit to that first repo, which also no longer show up.
> 
> Comments, Issues, branches, releases and Pull Requests in any form are entirely gone from the old 'Following'-Tab.
> 
> PS: **Even activity in my own repositories is entirely gone**
> 

Is "following" page still showing all those information correctly?

**Reply by Pingger** [reply]
Again: No. The 'following' page is _supposedly_ the old homepage. But to reiterate again: Bascially everything that was there before "for you" was introduced is now entirely gone.

**Reply by mikekavouras** [reply]
@Pingger could you elaborate a bit more on what you expect to show that isn't showing? We haven't make _any_ changes to the `Following` feed. There was however an infra issue over the past few weeks causing events to take upwards of 75 minutes to appear. This has since been resolved.

For reference, this is a screenshot of a portion of my `Following` feed
<img width="650" alt="CleanShot 2022-08-01 at 12 04 18@2x" src="https://user-images.githubusercontent.com/627794/182191700-e54f19bb-14df-401a-bb87-54f60dbbf1d0.png">

**Reply by Pingger** [reply]
For example:
I am following the Projects "Netdata", "AppliedEnergistics 2", and about 30 others. And all I have this:

![image](https://user-images.githubusercontent.com/4761298/182192397-7cd4b049-d689-4444-bc8a-9dd0a3941643.png)

**Reply by mikekavouras** [reply]
Thanks for the screenshot. I'll take a look and see if this is an issue with events getting dropped.

**Reply by mikekavouras** [reply]
@Pingger I haven't dug into specifics, but it looks like we only create Following feed release events if you **a)** follow the release author, or **b)**, follow one of the commit authors included in the release, but not if you've starred the project. If you've starred the project you should see these events in your `For you` feed. This code hasn't changed since Nov 11, 2021.

Taking a look at your public profile I can see you're following 6 users. My guess is those 6 users aren't generating much feed activity, hence the almost-empty Following feed.

I'm happy to answer any follow-up questions.

**Reply by Pingger** [reply]
![image](https://user-images.githubusercontent.com/4761298/182203053-ef717301-79f1-4376-acce-b55dc0a97cef.png)
This is what the "Following"-page states. (when scrolling to the bottom)
This is also how it was before the "For You"-Page was introduced.
And I can guarantee, that at the very least releases of starred and watched Repos where displayed in the "Following"-Page.

**Reply by mikekavouras** [reply]
It looks like we included starred repo releases in the feed between ~Jun 02, 2021 and ~Aug 11, 2021. Otherwise they'll show based on those two follow conditions I mentioned above. I can dig a bit deeper into other repo-related events.

> everytime I go to 'github.com' I end up on that new 'For You'-Page and thus am a click further away from the old homepage.

This should respect your selection. I believe it's using local storage to track that information so it may not work with a private browsing session or with certain browser plugins.

**Reply by Pingger** [reply]
> If you've starred the project you should see these events in your `For you` feed.

Yes along with quite a bit of stuff I don't want.
My Selection:
![image](https://user-images.githubusercontent.com/4761298/182211704-c5a3b0f8-38d9-4ebb-a45a-dbc95bf53bce.png)
What I get:
![image](https://user-images.githubusercontent.com/4761298/182211743-5bbba532-4e12-40f6-bc96-c41b85c0fdf8.png)
No I don't follow these people, I don't even KNOW these people. No I'm not watching either repository nor do I have them starred.

So in the I end up at the same conclusion: I want my old News feed back, that contains what I want and not what I don't want.

Even unticking "Follows", "Stars" and "Repositories" does not remove those 2 entries!

And to quote another user: "I have left websites for less".

**Reply by mikekavouras** [reply]
We're actively working on some tooling to enable better curation ("don't show me stuff from this person" or "I don't want to see this type of event"). I expect that will solve some of this frustration. That said, let me know if you have any interest in a quick video chat. You can reach out to me directly at mikekavouras@github.com

**Reply by NitinIsFit** [reply]
Could developers add a feature just like the notification? Rn we receive notifications from those repos which we turn on, like we get a pull request notification when we turn on pull request notifications on a repository, so in the for you page case, we can do something like "show me when someone makes issues on this repo in for you tab", "show me when someone opens a pull request on this repo in for you tab", "show me when blah-blah user follows someone in the for you tab", show me when blah-blah user makes a discussion in the for you tab", "show me when a blah blah user follows someone in the for you tab", "dont show me when a blah-blah person stars a repo, i wanna turn it manually off", "dont notify me about this user in for you feed, i wanna manually turn it off", "dont notify me anything about this person, i wanna manually turn it off", "notify me everything this person does anything in for page"

**Reply by jfullstackdev** [reply]
there is already that feature, the only thing that will be added, as I believe, is the exclusion, like "I don't want to hear from this user" without blocking them, although there is already the "Ignore" for a specific repo

![image](https://user-images.githubusercontent.com/47092464/182535798-9512922c-fb76-4ebf-a6af-1d63eed6d17d.png)

![image](https://user-images.githubusercontent.com/47092464/182535835-16b14f95-a6be-400f-8ef9-2baf54168416.png)

**Reply by NitinIsFit** [reply]
> there is already that feature, the only thing that will be added, as I believe, is the exclusion, like "I don't want to hear from this user" without blocking them, although there is already the "Ignore" for a specific repo
> 
> ![image](https://user-images.githubusercontent.com/47092464/182535798-9512922c-fb76-4ebf-a6af-1d63eed6d17d.png)
> 
> ![image](https://user-images.githubusercontent.com/47092464/182535835-16b14f95-a6be-400f-8ef9-2baf54168416.png)
> 
> 

Is this function available for 'for you' page tho?

**Reply by jfullstackdev** [reply]
it's not directly on the `For you` of course, it's on the repo itself, simply click the link of the repo when you see that on `For you` and you can select those notifs

**Comment by twome** [comment]
The intersocial aspect of GitHub, which for now is the "default" interface and network for all code worldwide, and thus the most convenient place to conduct *intersocial* things like pull requests and issues, is literally the only reason to use it. Everything else can more easily and robustly be done with FOSS self-hosted alternatives. As soon as someone makes a federated *social* alternative to GitHub's proprietary dominance that's good enough to exceed the existing network effect, there's no more reason to use this closed-source, proprietary, for-profit product controlled by Microsoft and with a revolting history of human rights abuse.

**Comment by Burnett01** [comment]
nah it's great and they should keep it. I found interesting projects from followers that way. I like the "For you" page,

**Reply by Tricked-dev** [reply]
same i found so many cools repo's with it that i would have never found otherwise
